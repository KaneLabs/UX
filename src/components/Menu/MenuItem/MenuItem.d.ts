import React, { Component } from 'react';
declare class MenuItem extends Component {
    constructor(props: any);
    handleHover(toggle: any): void;
    _renderIcon(): React.DetailedReactHTMLElement<{
        size: number;
        color: string;
    }, HTMLElement>;
    _renderKeyboardCommand(): JSX.Element;
    _renderText(): JSX.Element;
    render(): JSX.Element;
}
export default MenuItem;
