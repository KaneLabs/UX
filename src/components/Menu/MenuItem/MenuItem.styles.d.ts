export default styles;
declare namespace styles {
    namespace container {
        const minHeight: number;
        const width: number;
        const minWidth: number;
        const alignItems: string;
        const flexDirection: string;
        const paddingVertical: number;
        const paddingHorizontal: number;
    }
    namespace title {
        const fontSize: number;
        const fontWeight: string;
        const lineHeight: number;
        const letterSpacing: number;
        const color: string;
    }
}
