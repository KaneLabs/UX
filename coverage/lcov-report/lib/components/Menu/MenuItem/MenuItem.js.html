
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for lib/components/Menu/MenuItem/MenuItem.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../index.html">All files</a> / <a href="index.html">lib/components/Menu/MenuItem</a> MenuItem.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">76.32% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>58/76</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>13/26</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">53.85% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>7/13</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span></td><td class="text"><pre class="prettyprint lang-js">var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");var _interopRequireWildcard=require("@babel/runtime/helpers/interopRequireWildcard");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _classCallCheck2=_interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));var _createClass2=_interopRequireDefault(require("@babel/runtime/helpers/createClass"));var _inherits2=_interopRequireDefault(require("@babel/runtime/helpers/inherits"));var _possibleConstructorReturn2=_interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));var _getPrototypeOf2=_interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));var _react=_interopRequireWildcard(require("react"));var _propTypes=_interopRequireDefault(require("prop-types"));var _Text=_interopRequireDefault(require("react-native-web/dist/exports/Text"));var _Platform=_interopRequireDefault(require("react-native-web/dist/exports/Platform"));var _View=_interopRequireDefault(require("react-native-web/dist/exports/View"));var _MenuItem=_interopRequireDefault(require("./MenuItem.styles"));var _Ripple=_interopRequireDefault(require("../../Ripple/Ripple.js"));var _Hoverable=_interopRequireDefault(require("../../Utils/Hoverable/Hoverable.js"));var _jsxFileName="/Users/kane/Lab/Eros/src/components/Menu/MenuItem/MenuItem.js";function _createSuper(Derived){var hasNativeReflectConstruct=_isNativeReflectConstruct();return function _createSuperInternal(){var Super=(0,_getPrototypeOf2.default)(Derived),result;<span class="missing-if-branch" title="else path not taken" >E</span>if(hasNativeReflectConstruct){var NewTarget=(0,_getPrototypeOf2.default)(this).constructor;result=Reflect.construct(Super,arguments,NewTarget);}else{<span class="cstat-no" title="statement not covered" >result=Super.apply(this,arguments);}</span>return(0,_possibleConstructorReturn2.default)(this,result);};}function _isNativeReflectConstruct(){<span class="missing-if-branch" title="if path not taken" >I</span>if(typeof Reflect==="undefined"||!Reflect.construct)<span class="cstat-no" title="statement not covered" >return false;<span class="missing-if-branch" title="if path not taken" >I</span>i</span>f(Reflect.construct.sham)<span class="cstat-no" title="statement not covered" >return false;<span class="missing-if-branch" title="else path not taken" >E</span>i</span>f(typeof Proxy==="function")return true;<span class="cstat-no" title="statement not covered" >try{<span class="cstat-no" title="statement not covered" >Boolean.prototype.valueOf.call(Reflect.construct(Boolean,[],<span class="fstat-no" title="function not covered" >fu</span>nction(){}));<span class="cstat-no" title="statement not covered" >r</span>eturn true;}</span>catch(e){<span class="cstat-no" title="statement not covered" >return false;}</span>}</span>var MenuItem=function(_Component){(0,_inherits2.default)(MenuItem,_Component);var _super=_createSuper(MenuItem);function MenuItem(props){var _this;(0,_classCallCheck2.default)(this,MenuItem);_this=_super.call(this,props);_this.state={stateBackgroundColor:null};return _this;}(0,_createClass2.default)(MenuItem,[{key:"handleHover",value:function <span class="fstat-no" title="function not covered" >handleHover(</span>toggle){var disabled=<span class="cstat-no" title="statement not covered" >this.props.disabled;</span>var implementedColor=<span class="cstat-no" title="statement not covered" >toggle?'rgba(0,0,0,.12)':null;<span class="cstat-no" title="statement not covered" ></span>if(disabled)<span class="cstat-no" title="statement not covered" >implementedColor=null;<span class="cstat-no" title="statement not covered" >t</span></span>his.setState({stateBackgroundColor:implementedColor});}</span>},{key:"_renderIcon",value:function <span class="fstat-no" title="function not covered" >_renderIcon(</span>){var icon=<span class="cstat-no" title="statement not covered" >this.props.icon;<span class="cstat-no" title="statement not covered" ></span>return _react.default.cloneElement(icon,{size:24,color:'rgba(0, 0, 0, 0.54)'});}</span>},{key:"_renderKeyboardCommand",value:function <span class="fstat-no" title="function not covered" >_renderKeyboardCommand(</span>){var keyboardCommand=<span class="cstat-no" title="statement not covered" >this.props.keyboardCommand;<span class="cstat-no" title="statement not covered" ></span>return _react.default.createElement(_react.default.Fragment,null,_react.default.createElement(_View.default,{style:{flex:1},__self:this,__source:{fileName:_jsxFileName,lineNumber:38,columnNumber:9}}),keyboardCommand);}</span>},{key:"_renderText",value:function _renderText(){var _this$props=this.props,disabled=_this$props.disabled,textStyle=_this$props.textStyle,text=_this$props.text,icon=_this$props.icon;return _react.default.createElement(_Text.default,{ellipsizeMode:_Platform.default.OS==='ios'?<span class="branch-0 cbranch-no" title="branch not covered" >'clip':</span>'tail',numberOfLines:1,style:[_MenuItem.default.title,{color:disabled?<span class="branch-0 cbranch-no" title="branch not covered" >'rgba(0,0,0,.50)':</span>'rgba(0,0,0,.87)',marginLeft:icon?<span class="branch-0 cbranch-no" title="branch not covered" >20:</span>0},textStyle],__self:this,__source:{fileName:_jsxFileName,lineNumber:49,columnNumber:7}},text);}},{key:"render",value:function render(){var _this2=this;var _this$props2=this.props,children=_this$props2.children,disabled=_this$props2.disabled,onPress=_this$props2.onPress,style=_this$props2.style,icon=_this$props2.icon,keyboardCommand=_this$props2.keyboardCommand;var stateBackgroundColor=this.state.stateBackgroundColor;return _react.default.createElement(_Hoverable.default,{onHoverIn:function <span class="fstat-no" title="function not covered" >onHoverIn(</span>){<span class="cstat-no" title="statement not covered" >return _this2.handleHover(true);}</span>,onHoverOut:function <span class="fstat-no" title="function not covered" >onHoverOut(</span>){<span class="cstat-no" title="statement not covered" >return _this2.handleHover(false);}</span>,__self:this,__source:{fileName:_jsxFileName,lineNumber:73,columnNumber:7}},_react.default.createElement(_Ripple.default,{disabled:disabled,onPress:onPress,style:[_MenuItem.default.container,{backgroundColor:stateBackgroundColor},style],__self:this,__source:{fileName:_jsxFileName,lineNumber:77,columnNumber:9}},icon?<span class="branch-0 cbranch-no" title="branch not covered" >this._renderIcon():</span>null,children||this._renderText(),keyboardCommand?<span class="branch-0 cbranch-no" title="branch not covered" >this._renderKeyboardCommand():</span>null));}}]);return MenuItem;}(_react.Component);MenuItem.propTypes={children:_propTypes.default.node,disabled:_propTypes.default.bool,onPress:_propTypes.default.func,style:_propTypes.default.oneOfType([_propTypes.default.object,_propTypes.default.array]),textStyle:_propTypes.default.oneOfType([_propTypes.default.object,_propTypes.default.array]),icon:_propTypes.default.node,keyboardCommand:_propTypes.default.node,text:_propTypes.default.string};MenuItem.defaultProps={disabled:false};var _default=MenuItem;exports.default=_default;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Sun Jul 11 2021 09:12:08 GMT-0600 (Mountain Daylight Time)
            </div>
        </div>
        <script src="../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../sorter.js"></script>
        <script src="../../../../block-navigation.js"></script>
    </body>
</html>
    