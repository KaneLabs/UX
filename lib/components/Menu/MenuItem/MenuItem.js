var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");var _interopRequireWildcard=require("@babel/runtime/helpers/interopRequireWildcard");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _classCallCheck2=_interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));var _createClass2=_interopRequireDefault(require("@babel/runtime/helpers/createClass"));var _inherits2=_interopRequireDefault(require("@babel/runtime/helpers/inherits"));var _possibleConstructorReturn2=_interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));var _getPrototypeOf2=_interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));var _react=_interopRequireWildcard(require("react"));var _propTypes=_interopRequireDefault(require("prop-types"));var _Text=_interopRequireDefault(require("react-native-web/dist/exports/Text"));var _Platform=_interopRequireDefault(require("react-native-web/dist/exports/Platform"));var _View=_interopRequireDefault(require("react-native-web/dist/exports/View"));var _MenuItem=_interopRequireDefault(require("./MenuItem.styles"));var _Ripple=_interopRequireDefault(require("../../Ripple/Ripple.js"));var _Hoverable=_interopRequireDefault(require("../../Utils/Hoverable/Hoverable.js"));var _jsxFileName="/Users/kane/Lab/Eros/src/components/Menu/MenuItem/MenuItem.js";function _createSuper(Derived){var hasNativeReflectConstruct=_isNativeReflectConstruct();return function _createSuperInternal(){var Super=(0,_getPrototypeOf2.default)(Derived),result;if(hasNativeReflectConstruct){var NewTarget=(0,_getPrototypeOf2.default)(this).constructor;result=Reflect.construct(Super,arguments,NewTarget);}else{result=Super.apply(this,arguments);}return(0,_possibleConstructorReturn2.default)(this,result);};}function _isNativeReflectConstruct(){if(typeof Reflect==="undefined"||!Reflect.construct)return false;if(Reflect.construct.sham)return false;if(typeof Proxy==="function")return true;try{Boolean.prototype.valueOf.call(Reflect.construct(Boolean,[],function(){}));return true;}catch(e){return false;}}var MenuItem=function(_Component){(0,_inherits2.default)(MenuItem,_Component);var _super=_createSuper(MenuItem);function MenuItem(props){var _this;(0,_classCallCheck2.default)(this,MenuItem);_this=_super.call(this,props);_this.state={stateBackgroundColor:null};return _this;}(0,_createClass2.default)(MenuItem,[{key:"handleHover",value:function handleHover(toggle){var disabled=this.props.disabled;var implementedColor=toggle?'rgba(0,0,0,.12)':null;if(disabled)implementedColor=null;this.setState({stateBackgroundColor:implementedColor});}},{key:"_renderIcon",value:function _renderIcon(){var icon=this.props.icon;return _react.default.cloneElement(icon,{size:24,color:'rgba(0, 0, 0, 0.54)'});}},{key:"_renderKeyboardCommand",value:function _renderKeyboardCommand(){var keyboardCommand=this.props.keyboardCommand;return _react.default.createElement(_react.default.Fragment,null,_react.default.createElement(_View.default,{style:{flex:1},__self:this,__source:{fileName:_jsxFileName,lineNumber:38,columnNumber:9}}),keyboardCommand);}},{key:"_renderText",value:function _renderText(){var _this$props=this.props,disabled=_this$props.disabled,textStyle=_this$props.textStyle,text=_this$props.text,icon=_this$props.icon;return _react.default.createElement(_Text.default,{ellipsizeMode:_Platform.default.OS==='ios'?'clip':'tail',numberOfLines:1,style:[_MenuItem.default.title,{color:disabled?'rgba(0,0,0,.50)':'rgba(0,0,0,.87)',marginLeft:icon?20:0},textStyle],__self:this,__source:{fileName:_jsxFileName,lineNumber:49,columnNumber:7}},text);}},{key:"render",value:function render(){var _this2=this;var _this$props2=this.props,children=_this$props2.children,disabled=_this$props2.disabled,onPress=_this$props2.onPress,style=_this$props2.style,icon=_this$props2.icon,keyboardCommand=_this$props2.keyboardCommand;var stateBackgroundColor=this.state.stateBackgroundColor;return _react.default.createElement(_Hoverable.default,{onHoverIn:function onHoverIn(){return _this2.handleHover(true);},onHoverOut:function onHoverOut(){return _this2.handleHover(false);},__self:this,__source:{fileName:_jsxFileName,lineNumber:73,columnNumber:7}},_react.default.createElement(_Ripple.default,{disabled:disabled,onPress:onPress,style:[_MenuItem.default.container,{backgroundColor:stateBackgroundColor},style],__self:this,__source:{fileName:_jsxFileName,lineNumber:77,columnNumber:9}},icon?this._renderIcon():null,children||this._renderText(),keyboardCommand?this._renderKeyboardCommand():null));}}]);return MenuItem;}(_react.Component);MenuItem.propTypes={children:_propTypes.default.node,disabled:_propTypes.default.bool,onPress:_propTypes.default.func,style:_propTypes.default.oneOfType([_propTypes.default.object,_propTypes.default.array]),textStyle:_propTypes.default.oneOfType([_propTypes.default.object,_propTypes.default.array]),icon:_propTypes.default.node,keyboardCommand:_propTypes.default.node,text:_propTypes.default.string};MenuItem.defaultProps={disabled:false};var _default=MenuItem;exports.default=_default;