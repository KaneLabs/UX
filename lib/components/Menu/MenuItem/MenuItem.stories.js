var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _react=_interopRequireDefault(require("react"));var _Text=_interopRequireDefault(require("react-native-web/dist/exports/Text"));var _=require("../../..");var _Header=_interopRequireDefault(require("../../../storybook/components/Header"));var _Container=_interopRequireDefault(require("../../../storybook/components/Container"));var _storiesOf=require("../../../storybook/helpers/storiesOf");var _this=this,_jsxFileName="/Users/kane/Lab/Eros/src/components/Menu/MenuItem/MenuItem.stories.js";var _default=(0,_storiesOf.storiesOf)('Components|Menu/MenuItem',module).addParameters({jest:['MenuItem']}).add('Simple',function(){return _react.default.createElement(_Container.default,{__self:_this,__source:{fileName:_jsxFileName,lineNumber:15,columnNumber:5}},_react.default.createElement(_Header.default,{title:"Menu Item",__self:_this,__source:{fileName:_jsxFileName,lineNumber:16,columnNumber:7}}),_react.default.createElement(_.Paper,{radius:6,style:{width:324},__self:_this,__source:{fileName:_jsxFileName,lineNumber:18,columnNumber:7}},_react.default.createElement(_.MenuItem,{text:"Menu Item 1",__self:_this,__source:{fileName:_jsxFileName,lineNumber:19,columnNumber:9}}),_react.default.createElement(_.MenuItem,{text:"Menu Item 2",__self:_this,__source:{fileName:_jsxFileName,lineNumber:20,columnNumber:9}}),_react.default.createElement(_.Divider,{__self:_this,__source:{fileName:_jsxFileName,lineNumber:21,columnNumber:9}}),_react.default.createElement(_.MenuItem,{disabled:true,text:"Menu Item 3",__self:_this,__source:{fileName:_jsxFileName,lineNumber:22,columnNumber:9}}),_react.default.createElement(_.MenuItem,{text:"Menu Item 4",__self:_this,__source:{fileName:_jsxFileName,lineNumber:23,columnNumber:9}})));}).add('with icon',function(){return _react.default.createElement(_Container.default,{__self:_this,__source:{fileName:_jsxFileName,lineNumber:28,columnNumber:5}},_react.default.createElement(_Header.default,{title:"Menu icons",__self:_this,__source:{fileName:_jsxFileName,lineNumber:29,columnNumber:7}}),_react.default.createElement(_.Paper,{radius:6,style:{width:324},__self:_this,__source:{fileName:_jsxFileName,lineNumber:31,columnNumber:7}},_react.default.createElement(_.MenuItem,{text:"Preview",icon:_react.default.createElement(_.Icon,{name:"assignment-turned-in",__self:_this,__source:{fileName:_jsxFileName,lineNumber:34,columnNumber:17}}),__self:_this,__source:{fileName:_jsxFileName,lineNumber:32,columnNumber:9}}),_react.default.createElement(_.MenuItem,{text:"Share",icon:_react.default.createElement(_.Icon,{name:"person-add",__self:_this,__source:{fileName:_jsxFileName,lineNumber:36,columnNumber:38}}),__self:_this,__source:{fileName:_jsxFileName,lineNumber:36,columnNumber:9}}),_react.default.createElement(_.MenuItem,{text:"Get Link",icon:_react.default.createElement(_.Icon,{name:"link",__self:_this,__source:{fileName:_jsxFileName,lineNumber:37,columnNumber:41}}),__self:_this,__source:{fileName:_jsxFileName,lineNumber:37,columnNumber:9}}),_react.default.createElement(_.Divider,{__self:_this,__source:{fileName:_jsxFileName,lineNumber:38,columnNumber:9}}),_react.default.createElement(_.MenuItem,{text:"Preview",icon:_react.default.createElement(_.Icon,{name:"content-copy",__self:_this,__source:{fileName:_jsxFileName,lineNumber:39,columnNumber:40}}),__self:_this,__source:{fileName:_jsxFileName,lineNumber:39,columnNumber:9}}),_react.default.createElement(_.MenuItem,{text:"Download",icon:_react.default.createElement(_.Icon,{name:"file-download",__self:_this,__source:{fileName:_jsxFileName,lineNumber:40,columnNumber:41}}),__self:_this,__source:{fileName:_jsxFileName,lineNumber:40,columnNumber:9}})));}).add('with keyboard',function(){return _react.default.createElement(_Container.default,{__self:_this,__source:{fileName:_jsxFileName,lineNumber:45,columnNumber:5}},_react.default.createElement(_Header.default,{title:"Menu keyboard commands",__self:_this,__source:{fileName:_jsxFileName,lineNumber:46,columnNumber:7}}),_react.default.createElement(_.Paper,{radius:6,style:{width:324},__self:_this,__source:{fileName:_jsxFileName,lineNumber:48,columnNumber:7}},_react.default.createElement(_.MenuItem,{text:"Cut",icon:_react.default.createElement(_.Icon,{name:"content-cut",__self:_this,__source:{fileName:_jsxFileName,lineNumber:51,columnNumber:17}}),keyboardCommand:_react.default.createElement(_Text.default,{style:{fontSize:12,color:'#6e6e6e'},__self:_this,__source:{fileName:_jsxFileName,lineNumber:53,columnNumber:13}},"\u2318X"),__self:_this,__source:{fileName:_jsxFileName,lineNumber:49,columnNumber:9}}),_react.default.createElement(_.MenuItem,{text:"Copy",icon:_react.default.createElement(_.Icon,{name:"content-copy",__self:_this,__source:{fileName:_jsxFileName,lineNumber:59,columnNumber:17}}),keyboardCommand:_react.default.createElement(_Text.default,{style:{fontSize:12,color:'#6e6e6e'},__self:_this,__source:{fileName:_jsxFileName,lineNumber:61,columnNumber:13}},"\u2318C"),__self:_this,__source:{fileName:_jsxFileName,lineNumber:57,columnNumber:9}}),_react.default.createElement(_.MenuItem,{text:"Paste",icon:_react.default.createElement(_.Icon,{name:"content-paste",__self:_this,__source:{fileName:_jsxFileName,lineNumber:67,columnNumber:17}}),keyboardCommand:_react.default.createElement(_Text.default,{style:{fontSize:12,color:'#6e6e6e'},__self:_this,__source:{fileName:_jsxFileName,lineNumber:69,columnNumber:13}},"\u2318V"),__self:_this,__source:{fileName:_jsxFileName,lineNumber:65,columnNumber:9}})));});exports.default=_default;