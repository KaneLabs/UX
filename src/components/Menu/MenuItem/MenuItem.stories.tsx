import React from 'react';
import { Text } from 'react-native';

import { MenuItem } from 'eros-ui/components/Menu';
import Paper from 'eros-ui/components/Paper';
import Divider from 'eros-ui/components/Divider';
import Icon from 'eros-ui/components/Icon';

import Header from 'eros-ui-storybook/components/Header';
import Container from 'eros-ui-storybook/components/Container';
import { storiesOf } from 'eros-ui-storybook/helpers/storiesOf';

export default storiesOf('Components|Menu/MenuItem', module)
  .addParameters({ jest: ['MenuItem'] })
  .add('Simple', () => (
    <Container>
      <Header title="Menu Item" />

      <Paper style={{ width: 324 }}>
        <MenuItem text="Menu Item 1" />
        <MenuItem text="Menu Item 2" />
        <Divider />
        <MenuItem disabled text="Menu Item 3" />
        <MenuItem text="Menu Item 4" />
      </Paper>
    </Container>
  ))
  .add('with icon', () => (
    <Container>
      <Header title="Menu icons" />

      <Paper style={{ width: 324 }}>
        <MenuItem text="Preview" icon={<Icon name="assignment-turned-in" />} />
        <MenuItem text="Share" icon={<Icon name="person-add" />} />
        <MenuItem text="Get Link" icon={<Icon name="link" />} />
        <Divider />
        <MenuItem text="Preview" icon={<Icon name="content-copy" />} />
        <MenuItem text="Download" icon={<Icon name="file-download" />} />
      </Paper>
    </Container>
  ))
  .add('with keyboard', () => (
    <Container>
      <Header title="Menu keyboard commands" />

      <Paper style={{ width: 324 }}>
        <MenuItem
          text="Cut"
          icon={<Icon name="content-cut" />}
          keyboardCommand={
            <Text style={{ fontSize: 12, color: '#6e6e6e' }}>&#8984;X</Text>
          }
        />

        <MenuItem
          text="Copy"
          icon={<Icon name="content-copy" />}
          keyboardCommand={
            <Text style={{ fontSize: 12, color: '#6e6e6e' }}>&#8984;C</Text>
          }
        />

        <MenuItem
          text="Paste"
          icon={<Icon name="content-paste" />}
          keyboardCommand={
            <Text style={{ fontSize: 12, color: '#6e6e6e' }}>&#8984;V</Text>
          }
        />
      </Paper>
    </Container>
  ));
