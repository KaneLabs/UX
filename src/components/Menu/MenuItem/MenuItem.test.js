import React from 'react';
import renderer from 'react-test-renderer';
import MenuItem from './MenuItem';

test('MenuItem Renders', () => {
  const tree = renderer.create(<MenuItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
