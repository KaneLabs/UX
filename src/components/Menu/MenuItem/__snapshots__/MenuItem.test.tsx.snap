// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem Renders 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    accessible={true}
    displayUntilPressOut={true}
    focusable={true}
    onClick={[Function]}
    onLayout={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onRippleAnimation={[Function]}
    onStartShouldSetResponder={[Function]}
    rippleCentered={false}
    rippleColor="rgba(0, 0, 0, .87)"
    rippleDuration={400}
    rippleFades={true}
    rippleOpacity={0.3}
    rippleSequential={false}
    rippleSize={0}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": null,
        "flexDirection": "row",
        "minHeight": 32,
        "minWidth": 124,
        "paddingHorizontal": 24,
        "paddingVertical": 12,
        "width": 324,
      }
    }
    testID="mb-ripple"
    useNativeDriver={false}
  >
    <Text
      ellipsizeMode="clip"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#000000",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.44,
            "lineHeight": 24,
          },
          Object {
            "color": "rgba(0,0,0,.87)",
            "marginLeft": 0,
          },
          undefined,
        ]
      }
    />
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "overflow": "hidden",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 10,
          },
          Object {
            "borderRadius": 0,
          },
        ]
      }
    />
  </View>
</View>
`;
